Formula/shadowenv.rb

```diff
@@ -0,0 +1,21 @@
+class Shadowenv < Formula
+  desc "Reversible directory-local environment variable manipulations"
+  homepage "https://shopify.github.io/shadowenv/"
+  url "https://github.com/Shopify/shadowenv/archive/0.3.1.tar.gz"
+  sha256 "303d0537c37fd02f6c75e6c65fa7662c5a9d8c939548cd1c02b1b1fb746cd560"
+
+  depends_on "rustup-init" => :build
+
+  def install
+    system "rustup-init", "-y"
+    ENV["PATH"] = "#{ENV["CARGO_HOME"]}/bin:#{ENV["PATH"]}"
+    system "rustup", "toolchain", "install", "nightly-2018-12-26"
+    system "cargo", "+nightly-2018-12-26", "install", "--root", prefix, "--path", "."
+    man1.install "#{buildpath}/man/man1/shadowenv.1"
+    man5.install "#{buildpath}/man/man5/shadowlisp.5"
+  end
+
+  test do
+    assert_equal "shadowenv 0.3.1", shell_output("#{bin}/shadowenv -V").chomp
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.